### PR TITLE
F add single & edit record page & function

### PR DESCRIPTION
**Pull request by NilsMester**

### CODE REVIEW

File: frontend/src/contexts/RecordContextProvider.js [hunk 1]
```diff
@@ -28,8 +28,19 @@ export default function RecordContextProvider({children}) {
...
+        removeRecord(id, token)
+            .then(() => setRecords(records.filter((record) => record.id !== id)))
+            .catch(console.log);
+
+    const editRecord = (id, titel, recordLink, tagsList, description, publicStaus) =>{
+        updateRecord(id, titel, recordLink, tagsList, description, publicStaus, token)
+            .then((updatedRecord) => setRecords([...records.filter((record) => record.id !== updatedRecord.id), updatedRecord]))
+            .catch(console.log)
+    }
+
```
Comment (Asamya) [inline frontend/src/contexts/RecordContextProvider.js:41]: I think you only need the `.catch` statements if you have no security token. Then you would catch the error in the `console.log`. Now it is the security context that catches the error.

**Review by Asamya — APPROVED** [review]
Nice work. Only one minor comment.

**Review by kainary91 — APPROVED** [review]

File: frontend/src/service/RecordService.js [hunk 2]
```diff
@@ -10,9 +10,18 @@ const header = (token) => ({
...
 export const addRecord = (titel, recordLink, tagsList, description, publicStaus, token) =>
     axios
 .post('/api/records', {titel, recordLink, description, publicStaus, tagsList}, header(token))
-.then((response) => response.data)
\ No newline at end of file
+.then((response) => response.data);
+
+export const removeRecord = (id, token) =>
+    axios
+    .delete('/api/recods/' + id, header(token));
```
Comment (christophersiem) [inline frontend/src/service/RecordService.js:22]:
```suggestion
    .delete('/api/records/' + id, header(token));
```

File: frontend/src/service/RecordService.js [hunk 3]
```diff
@@ -10,9 +10,18 @@ const header = (token) => ({
...
+
+export const updateRecord = (id, titel, recordLink, tagsList, description, publicStaus, token) =>
```
Comment (christophersiem) [inline frontend/src/service/RecordService.js:24]:
```suggestion
export const updateRecord = (id, titel, recordLink, tagsList, description, publicStatus, token) =>
```

File: frontend/src/service/RecordService.js [hunk 4]
```diff
@@ -10,9 +10,18 @@ const header = (token) => ({
 export const getUserRecordsList = (token) =>
     axios
         .get('api/records', header(token))
-        .then((response) => response.data)
+        .then((response) => response.data);
 
 export const addRecord = (titel, recordLink, tagsList, description, publicStaus, token) =>
     axios
 .post('/api/records', {titel, recordLink, description, publicStaus, tagsList}, header(token))
```
Comment (christophersiem) [inline frontend/src/service/RecordService.js:17]:
```suggestion
.post('/api/records', {titel, recordLink, description, publicStatus, tagsList}, header(token))
```

File: frontend/src/pages/singleRecordPage/SingleRecordPage.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    const {id} = useParams();
+    const history = useHistory();
+
+    const record = records.find((record) => record.id === id);
+    return !record ? null : (
+        <>
+            <Header titel={record.titel}/>
+            <Record record={record} actions={getActions()}/>
+        </>
+);
```
Comment (christophersiem) [inline frontend/src/pages/singleRecordPage/SingleRecordPage.js:21]: for better readability I'd prefer the following:

```suggestion
    return { record &&
        <>
            <Header titel={record.titel}/>
            <Record record={record} actions={getActions()}/>
        </>
};
```

File: frontend/src/pages/editRecordPage/EditRecordPage.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import RecordForm from "../../commons/RecordForm";
+import Header from "../../commons/Header";
+
+export default function EditIdeaPage() {
+    const {editRecord, records} = useContext(RecordContext);
+    const history = useHistory();
+    const {id} = useParams();
+    const record = records.find((record) => record.id === id);
+
+    return !record ? null : (
```
Comment (christophersiem) [inline frontend/src/pages/editRecordPage/EditRecordPage.js:13]: see comment below in SingleRecordPage.js

**Review by christophersiem — COMMENTED** [review]
well done 🛴

**Pull request merged**